prefect_kubernetes/jobs.py

```diff
@@ -452,11 +427,36 @@ async def wait_for_completion(self):
                 )
             elif v1_job_status.status.succeeded:
                 self._completed = True
+                await self._retrieve_pod_logs(v1_job_status)
```

Comment:
This will only grab logs when the pod _succeeds_, but we also want to do it when the pod has _failed_ too (the branch of the `if` just above this).